src/Persistent_cohomology/include/gudhi/Persistent_cohomology.h

```diff
@@ -571,6 +571,7 @@ class Persistent_cohomology {
 
   void write_output_diagram(std::string diagram_name) {
     std::ofstream diagram_out(diagram_name.c_str());
+    diagram_out.exceptions(diagram_out.failbit);
```

Comment:
I did not know this one. Will it be catched at python level ?

Reply:
It gets converted to a python exception (because I added `except +` in the pxd file). Then it is up to the user to catch the exception if they call write_persistence_diagram with a bad file name, as it would be if they just called `open`.